File: CodeWars/8kyu/JavaScript/006-filling-an-array-part-1.js
// https://www.codewars.com/kata/571d42206414b103dc0006a1
// Write a function that produces an array with the numbers 0 to N-1 in it.
// arr(5) // => [0,1,2,3,4]

function arr(n) {
    let result = [];

    for (let i = 0; i < n; i++) {
        result.push(i)
    }

    return result;
}

function arr(n) {
    return Array.from({length: n}).map((_, i) => i);
}

console.log(arr(5));

File: CodeWars/8kyu/Python/006-filling-an-array-part-1.py
# https://www.codewars.com/kata/571d42206414b103dc0006a1
# Write a function that produces an array with the numbers 0 to N-1 in it.
# For example, the following code will result in an array containing the numbers 0 to 4:
# arr(5) // => [0,1,2,3,4]

def arr(n=0):
    result = []

    if n > 0:
        for i in range(n):
            result.append(i)
    return result


def arr(n=0): 
    return list(range(n))


def arr(n=0): 
    return [i for i in range(n)]

    
print(arr(4), [0, 1, 2, 3])
print(arr(0), [])
print(arr(), [])
